File: apps/android/feature/ai/domain/src/main/java/com/scribblefit/feature/ai/domain/model/FeedItem.kt
package com.scribblefit.feature.ai.domain.model

/**
 * Represents an item in the Intelligent Canvas feed.
 */
sealed class FeedItem {
    abstract val id: String
    abstract val timestamp: Long

    /**
     * An AI-generated prompt or suggestion.
     */
    data class Prompt(
        override val id: String,
        override val timestamp: Long,
        val text: String,
        val emoji: String,
        val type: SuggestionType
    ) : FeedItem()

    /**
     * A raw user entry (text or voice).
     */
    data class Scribble(
        override val id: String,
        override val timestamp: Long,
        val rawText: String,
        val status: ScribbleStatus
    ) : FeedItem()

    /**
     * A parsed, structured workout ready for confirmation.
     */
    data class Confirmation(
        override val id: String,
        override val timestamp: Long,
        val workout: ParsedWorkout,
        val scribbleId: String
    ) : FeedItem()

    /**
     * A micro-insight (e.g., "New PR!").
     */
    data class Insight(
        override val id: String,
        override val timestamp: Long,
        val text: String,
        val emoji: String
    ) : FeedItem()
}

enum class ScribbleStatus {
    PENDING, PROCESSING, FAILED, COMPLETED
}

File: apps/ios/ScribbleFit/Sources/Features/Canvas/Domain/Models/FeedItem.swift
import Foundation

public enum FeedItem: Sendable, Identifiable {
    case prompt(PromptItem)
    case scribble(ScribbleItem)
    case confirmation(ConfirmationItem)
    case insight(InsightItem)

    public var id: String {
        switch self {
        case .prompt(let item): return item.id
        case .scribble(let item): return item.id
        case .confirmation(let item): return item.id
        case .insight(let item): return item.id
        }
    }

    public var timestamp: Date {
        switch self {
        case .prompt(let item): return item.timestamp
        case .scribble(let item): return item.timestamp
        case .confirmation(let item): return item.timestamp
        case .insight(let item): return item.timestamp
        }
    }
}

public struct PromptItem: Identifiable, Codable, Sendable {
    public let id: String
    public let timestamp: Date
    public let text: String
    public let emoji: String
    public let type: SuggestionType
}

public struct ScribbleItem: Identifiable, Sendable {
    public let id: String
    public let timestamp: Date
    public let rawText: String
    public let status: ScribbleStatus
}

public struct ConfirmationItem: Identifiable, Codable, Sendable {
    public let id: String
    public let timestamp: Date
    public let workout: ParsedWorkout
    public let scribbleId: String
}

public struct InsightItem: Identifiable, Codable, Sendable {
    public let id: String
    public let timestamp: Date
    public let text: String
    public let emoji: String
}

public enum ScribbleStatus: Sendable {
    case pending, processing, failed, completed
}
